### PR TITLE
`azurerm_healthbot` - use LRO operation for bots update

### PR DESCRIPTION
**Pull request by neil-yechenwei**
Recently I found test case about HealthBot is failed. After investigated, I found HealthBot.Update API always returns 201 after the resource is updated successfully but the [rest api spec](https://github.com/Azure/azure-rest-api-specs/blob/d70ba571ddb519a63aa2ad96a96faff04647f81c/specification/healthbot/resource-manager/Microsoft.HealthBot/stable/2022-08-08/healthbot.json#L74) indicates service API should return 200 after updated successfully. So I assume we have to make this fix to workaround it for now.

image

### CODE REVIEW

File: internal/services/bot/bot_healthbot_resource.go [hunk 1]
```diff
@@ -161,8 +162,11 @@ func resourceHealthbotServiceUpdate(d *pluginsdk.ResourceData, meta interface{})
 		payload.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
 	}
 
-	if _, err := client.BotsUpdate(ctx, *id, payload); err != nil {
-		return fmt.Errorf("updating %s: %+v", id, err)
+	if resp, err := client.BotsUpdate(ctx, *id, payload); err != nil {
+		// update check logic once the issue https://github.com/Azure/azure-rest-api-specs/issues/22791 is fixed
+		if !response.WasStatusCode(resp.HttpResponse, http.StatusCreated) {
+			return fmt.Errorf("updating %s: %+v", id, err)
+		}
```
Comment (tombuildsstuff) [inline internal/services/bot/bot_healthbot_resource.go:169]: If this is a 201 that would indicate that this is an LRO, so there should be something [we can poll on using the pollers within `go-azure-sdk` here](https://github.com/hashicorp/go-azure-sdk/blob/6cdea3ca229b352fa45cf3efcc2ece9b0eaca92a/sdk/client/resourcemanager/poller.go#L12)?

Comment (neil-yechenwei) [reply internal/services/bot/bot_healthbot_resource.go:169]: As service team has submitted this [PR](https://github.com/Azure/azure-rest-api-specs/pull/22908) to fix this issue but it hasn't been merged, so I intend to update this PR accordingly once this PR is merged.

Comment (katbyte) [reply internal/services/bot/bot_healthbot_resource.go:169]: @neil-yechenwei - the pr linked here has been merged

Comment (neil-yechenwei) [reply internal/services/bot/bot_healthbot_resource.go]: Thanks @katbyte for reminder. I updated PR. 

image

Comment (neil-yechenwei) [reply internal/services/bot/bot_healthbot_resource.go]: @katbyte , kindly ping. Is it fine to merge this PR?

**Review by katbyte — APPROVED** [review]
LGTM 🤖

**Pull request merged**